### ISSUE TITLE
[bug] Ninja as Cmake generator is not picked up

### ISSUE THREAD

**Issue by Flow-It** [conan-io/conan#7069]
### Environment Details (include every applicable attribute)
  * Operating System+version: Arch Linux
  * Compiler+version: GCC 9.3 and GCC 10.1
  * Conan version: 1.25.1
  * Python version: 3.8.3

### Steps to reproduce (Include if Applicable)
Add `CONAN_CMAKE_GENERATOR = Ninja` to conan.conf. Make sure its the only like where this exact env var is set.
Run conan create for any Cmake project.

### Logs (Executed commands with output) (Include/Attach if Applicable)
The Cmake command is
`cmake -G "Unix Makefiles" [...]`

Am I missing something or why is make used instead of ninja when the environment variable is set?

**Comment by memsharded** [comment]
Seems to work fine here (windows):

```
$ conan new hello/0.1 -s
$ set CONAN_CMAKE_GENERATOR=Ninja
$ set CONAN_PRINT_RUN_COMMANDS=1
$ conan create .

> .... && cmake -G "Ninja" ...
```

Can you please run the above commands (with ``export`` in your case) in your system and paste output here?

**Comment by memsharded** [comment]
Sorry, nevermind, yes, I see, when using it from the conan.conf, it is not being used. Need to check more thoroughly. Thanks for reporting this.

**Comment by memsharded** [comment]
Ok, I see where the problem is. The env-vars follow syntax of env-vars, while conan.conf is doing more of a ``ini`` file. In the ``conan.conf`` you should have:

```ini
[general]
...
cmake_generator = Ninja
```

set it with
```bash
$ conan config set general.cmake_generator=Ninja
```

Have a look to: https://docs.conan.io/en/latest/reference/config_files/conan.conf.html

In any case, please try to use env-vars better from profiles (using profiles is very recommended), instead of in ``conan.conf``. These "build-specific" configuration belongs more to the profiles, and it is likely than in Conan 2.0 we will remove them from ``conan.conf``.

**Comment by Flow-It** [comment]
Thanks, that did the trick. Whatever your plans are, please don't forget to update the configuration. That's how I started it using it this way: https://docs.conan.io/en/latest/integrations/build_system/ninja.html

**Comment by memsharded** [comment]
Yes, this should be clarified in the docs. Moving the issue there, thanks for the feedback!
